agent/stats/envoy_prometheus_stats.go

```diff
@@ -129,32 +120,22 @@ func (envoyPrometheusStatsHandler *EnvoyPrometheusStatsHandler) HandleStats(resW
 	log.Debugf("Stats request took: %vms", duration.Milliseconds())
 
 	if err != nil {
-		errorMsg := fmt.Sprintf("unable to reach Envoy Admin: %s", err)
-		log.Error(errorMsg)
-		// Valid status code range is [100, 600)
-		if statsResponse != nil && statsResponse.StatusCode < 600 && statsResponse.StatusCode >= 100 {
-			http.Error(resWriter, errorMsg, statsResponse.StatusCode)
-		} else {
-			http.Error(resWriter, errorMsg, http.StatusInternalServerError)
-		}
-		return
-	}
-
-	if statsResponse == nil {
-		resWriter.WriteHeader(http.StatusNoContent)
-		log.Debug("empty response from Envoy stat endpoint")
+		log.Errorf("Call to fetch stats from Envoy admin failed: %s", err)
+		http.Error(resWriter, "Failed to fetch stats from Envoy", http.StatusInternalServerError)
 		return
 	}
 
 	if statsResponse.StatusCode != http.StatusOK {
-		http.Error(resWriter, fmt.Sprintf("Envoy response not OK, error code: %v", statsResponse.StatusCode), statsResponse.StatusCode)
+		log.Errorf("Envoy stats response status code not OK: %v", statsResponse.Status)
+		http.Error(resWriter, "Failed to fetch stats from Envoy", http.StatusInternalServerError)
```

Comment:
Responding with InternalServerError in all failure cases as, we don't need to leak internal details of interaction with Envoy admin to the client (ecs agent).

Reply:
I am fine with the change.

Just posting some background here. In the early discussion when we implement this, we wanted to "pass the original return code through to the client. Save the 500s for when the Agent itself has an unrecoverable error".

The clients may not just be ECS agent, if the customer start using SC agent stats directly and not using the Admin interface, they may see different response from the stats endpoints (Envoy vs SC agent).

Reply:
I think the idea is to abstract the Envoy interface behind the SC agent's management interface and the client doesn't need to be aware of the internals. I know it is possible currently to directly query Envoy but, that is not expected by the client and may change in future.